app/src/test/java/org/simple/clinic/user/finduser/UserLookupTest.kt

```diff
@@ -80,6 +86,60 @@ class UserLookupTest {
     assertThat(result).isEqualTo(UnexpectedError)
   }
 
+  @Test
+  fun `when the find user by phone number call succeeds, return the fetched data`() {
+    // given
+    val userUuid = UUID.fromString("70036d9e-6b5e-4166-9682-594647c32f26")
+    val userStatus = UserStatus.WaitingForApproval
+    whenever(registrationApi.findUserByPhoneNumber(FindUserRequest(phoneNumber)))
+        .doReturn(FakeCall.success(FindUserResponse(Body(userUuid, userStatus))))
+
+    // when
+    val result = findUserWithPhoneNumber.find(phoneNumber)
+
+    // then
+    assertThat(result).isEqualTo(Found(userUuid, userStatus))
+  }
+
+  @Test
+  fun `when the find user by phone number call returns 404, return the not found result`() {
+    // given
+    whenever(registrationApi.findUserByPhoneNumber(FindUserRequest(phoneNumber)))
+        .doReturn(FakeCall.error("", responseCode = 404))
+
+    // when
+    val result = findUserWithPhoneNumber.find(phoneNumber)
+
+    // then
+    assertThat(result).isEqualTo(NotFound)
+  }
+
+  @Test
+  fun `when the find user by phone number call fails with a network error, return the network error result`() {
+    // given
+    whenever(registrationApi.findUserByPhoneNumber(FindUserRequest(phoneNumber)))
+        .doReturn(FakeCall.failure(SocketTimeoutException()))
+
+    // when
+    val result = findUserWithPhoneNumber.find(phoneNumber)
+
+    // then
+    assertThat(result).isEqualTo(NetworkError)
+  }
+
+  @Test
+  fun `when the find user by phone number call returns a server error, return the other error result`() {
```

Comment:
Can you improve the description of this test a little to specify the exact error we expect?